third_party/intel/lib/TritonIntelGPUToLLVM/ArithOpsToLLVM.cpp

```diff
@@ -0,0 +1,85 @@
+#include "PatternTritonGPUOpToLLVM.h"
+
+#include "mlir/Conversion/LLVMCommon/TypeConverter.h"
+#include "mlir/IR/BuiltinTypes.h"
+
+using namespace mlir;
+
+/// Custom lowering for converting arith ops to LLVMIR dialect.
+/// These ops lowering ONLY works for the advanced path.
+/// This is a temporary solution until we have done upstreaming or have other
+/// proper lowering strategy for these ops.
+
+namespace {
+// FIXME: Remove this lowering when upstream ConstantOpLowering has such
+// lowering.
+class ArithConstantOpLowering
+    : public ConvertTritonGPUOpToLLVMPattern<mlir::arith::ConstantOp> {
+  using ConvertTritonGPUOpToLLVMPattern<
+      mlir::arith::ConstantOp>::ConvertTritonGPUOpToLLVMPattern;
+  LogicalResult
+  matchAndRewrite(mlir::arith::ConstantOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Location loc = op->getLoc();
+    auto srcType = dyn_cast<ShapedType>(op.getType());
+    if (!srcType || srcType.getNumElements() == 1)
+      return failure();
+
+    // Only handle constant with tensor types.
+    if (!isa<RankedTensorType>(srcType))
+      return failure();
+
+    Type dstType = getTypeConverter()->convertType(srcType);
+    if (!dstType)
+      return failure();
+
+    auto dstElementsAttr = dyn_cast<DenseElementsAttr>(op.getValue());
+    if (!dstElementsAttr)
+      return failure();
```

Comment:
```suggestion
    auto dstElementsAttr = dyn_cast<DenseElementsAttr>(op.getValue());
    if (!dstElementsAttr || !dstElementsAttr.isSplat())
      return failure();
```
This will only work if the constant is a splat, e.g., it won't work for:

```
%0 = arith.constant dense[0, 1, 2, 3, ...] : tensor<128xi32>
```

Reply:
Updated.